File: server/php/src/HostedCheckout/CreateHostedCheckoutBasicDto.php
<?php
namespace MyApp\HostedCheckout;
use JsonSerializable;

/**
 * Basic Dto for Create Hosted Checkout
 */
class CreateHostedCheckoutBasicDto implements JsonSerializable
{
    private $amount;
    private $currency;
    private $redirectUrl;

    public function getAmount() {
        return $this->amount;
    }

    public function setAmount($amount) {
        $this->amount = $amount;
    }

    public function getCurrency() {
        return $this->currency;
    }

    public function setCurrency($currency) {
        $this->currency = $currency;
    }

    public function getRedirectUrl() {
        return $this->redirectUrl;
    }

    public function setRedirectUrl($redirectUrl) {
        $this->redirectUrl = $redirectUrl;
    }

    public function jsonSerialize(): mixed
    {
        return [
            'amount' => $this->amount,
            'currency' => $this->currency,
            'redirectUrl' => $this->redirectUrl
        ];
    }
}

File: server/ruby/app/hosted_checkout/create_hosted_checkout_basic_dto.rb
class CreateHostedCheckoutBasicDto
  attr_accessor :amount, :currency, :redirect_url

  def initialize(amount, currency, redirect_url)
    @amount = amount
    @currency = currency
    @redirect_url = redirect_url
  end

  def to_h
    {
      amount: @amount,
      currency: @currency,
      redirectUrl: @redirect_url
    }
  end
end
